### PR TITLE
feat: Add hubspot destination connector

### PR DESCRIPTION
**Pull request by karthik-sivadas**
## Description
feat: Add hubspot destination connector

## Type of Change
- [x] New Connector (Destination or Source Connector)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would impact existing functionality)
- [ ] Documentation update
- [ ] Chore
  
## How Has This Been Tested?
Wrote specs and tested it locally

## Checklist:
- [x] Ensure a branch name is prefixed with `feature`, `bugfix`, `hotfix`, `release` or `chore` followed by `/` and branch name e.g `feature/add-salesforce-connector`
- [x] Added the new connector in rollout.rb
- [x] Have you updated the version number of the gem?
- [x] Have you ensured that your changes for new connector are documented in the [docs repo](https://github.com/Multiwoven/docs) or relevant documentation files?
- [x] Have you updated the run time dependency in multiwoven-integrations.gemspec if new gems are added

### CODE REVIEW

File: lib/multiwoven/integrations/destination/hubspot/client.rb [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+        include Multiwoven::Integrations::Core
+
+        class Client < DestinationConnector
+          def check_connection(connection_config)
+            connection_config = connection_config.with_indifferent_access
+            initialize_client(connection_config)
+            authenticate_client
+            success_status
+          rescue StandardError => e
+            failure_status(e)
```
Comment (subintp) [inline lib/multiwoven/integrations/destination/hubspot/client.rb:18]: Log error message with stack trace

File: lib/multiwoven/integrations/destination/hubspot/client.rb [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+              process_record(stream, record)
+              write_success += 1
+            rescue StandardError => e
+              handle_exception("HUBSPOT:CRM:WRITE:EXCEPTION", "error", e)
+              write_failure += 1
+            end
+            tracking_message(write_success, write_failure)
+          end
+
+          def process_record(stream, record)
```
Comment (subintp) [inline lib/multiwoven/integrations/destination/hubspot/client.rb:58]: what is the use of this method ?

File: lib/multiwoven/integrations/destination/hubspot/client.rb [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+            send_data_to_hubspot(stream.name, record)
+          end
+
+          def send_data_to_hubspot(stream_name, record = {})
+            args = build_args(@action, stream_name, record)
+            eval("@client.crm.#{stream_name}.basic_api.#{@action}(simple_public_object_input_for_create: #{args})") # rubocop:disable Security/Eval, Style/EvalWithLocation
```
Comment (subintp) [inline lib/multiwoven/integrations/destination/hubspot/client.rb:64]: Not able to understand. Add a comment on how this line works

**Review by subintp — CHANGES_REQUESTED** [review]

File: lib/multiwoven/integrations/destination/hubspot/client.rb (hunk 2 above)
Comment (karthik-sivadas) [reply lib/multiwoven/integrations/destination/hubspot/client.rb:58]: No special use will remove this

File: lib/multiwoven/integrations/destination/hubspot/client.rb (hunk 3 above)
Comment (karthik-sivadas) [reply lib/multiwoven/integrations/destination/hubspot/client.rb]: Changed the eval

File: lib/multiwoven/integrations/destination/hubspot/config/catalog.json [hunk 4]
```diff
@@ -0,0 +1,351 @@
+{
+  "request_rate_limit": 100000,
```
Comment (afthabvp) [inline lib/multiwoven/integrations/destination/hubspot/config/catalog.json:2]: rate limit values are the real numbers for hubspot

**Review by afthabvp — APPROVED** [review]
LGTM

**Review by subintp — APPROVED** [review]

**Pull request merged**